Assets/_BForBoss/_Core/Scripts/EnergySystem/EnergySystemBehaviour.cs

```diff
@@ -11,6 +11,7 @@ public interface IEnergySystem
     {
         void Accrue(EnergyAccruementType accruementType, float multiplier = 1);
         void Expend(EnergyExpenseType expenseType, float multiplier = 1);
+        bool CanExpend(EnergyExpenseType expenseType, float multiplier = 1);
```

Comment:
could make this `bool TryExpend()`? then we don't have to do `if CanExpend() {Expend()}` all the time

Reply:
In this case, I needed to have 2 seperate methods. 

Check if you can activate.. 
then activate + all the other stuff that goes along + tweening + sound... 

then you have the other state - While it's activated (while it's in a loop).  Then check again if you can still expend, and then expend energy.

Reply:
I think especially in this more generic interface, I rather have every method be clear cut, concise, and does only ONE simple function. 

And then in the concrete classes, u can do "Try" if need be. (use interface as building blocks, as opposed to force the concrete class to implement multiple functionality based on method name).